Mlem/Views/Tabs/Inbox/Inbox View.swift

```diff
@@ -80,58 +117,70 @@ struct InboxView: View {
         self._messageTracker = StateObject(wrappedValue: newMessageTracker)
     }
     
-    // input state handling
-    // - current view
     @State var curTab: InboxTab = .all
     
-    // utility
-    @StateObject private var inboxTabNavigation: AnyNavigationPath<AppRoute> = .init()
-    @StateObject private var navigation: Navigation = .init()
-    
     var body: some View {
-        ScrollViewReader { scrollProxy in
-            // NOTE: there appears to be a SwiftUI issue with segmented pickers stacked on top of ScrollViews which causes the tab bar to appear fully transparent. The internet suggests that this may be a bug that only manifests in dev mode, so, unless this pops up in a build, don't worry about it. If it does manifest, we can either put the Picker *in* the ScrollView (bad because then you can't access it without scrolling to the top) or put a Divider() at the bottom of the VStack (bad because then the material tab bar doesn't show)
-            NavigationStack(path: $inboxTabNavigation.path) {
-                contentView(scrollProxy: scrollProxy)
-                    .navigationTitle("Inbox")
-                    .navigationBarTitleDisplayMode(.inline)
-                    .navigationBarColor()
-                    .toolbar {
-                        ToolbarItemGroup(placement: .navigationBarTrailing) { ellipsisMenu }
-                    }
-                    .listStyle(PlainListStyle())
-                    .tabBarNavigationEnabled(.inbox, navigation)
-                    .handleLemmyViews()
-                    .environmentObject(inboxTabNavigation)
-                    .environmentObject(inboxTracker)
+        content
+            .navigationTitle("Inbox")
+            .navigationBarTitleDisplayMode(.inline)
+            .navigationBarColor()
+            .toolbar {
+                ToolbarItem(placement: .principal) {
+                    navBarTitle
+                        .opacity(scrollToTopAppeared ? 0 : 1)
+                        .animation(.easeOut(duration: 0.2), value: scrollToTopAppeared)
+                }
+                
+                ToolbarItemGroup(placement: .navigationBarTrailing) { ellipsisMenu }
             }
-            .handleLemmyLinkResolution(navigationPath: .constant(inboxTabNavigation))
-            .environment(\.navigationPathWithRoutes, $inboxTabNavigation.path)
-            .environment(\.navigation, navigation)
-            .environment(\.scrollViewProxy, scrollProxy)
-        }
-        .onChange(of: shouldFilterRead) { newValue in
-            Task(priority: .userInitiated) {
-                await handleShouldFilterReadChange(newShouldFilterRead: newValue)
+            .hoistNavigation {
+                withAnimation {
+                    scrollProxy?.scrollTo(scrollToTop)
+                }
+                return true
```

Comment:
Scroll-to-top doesn't appear to be working.

Reply:
🤦 I had tab navigation to `search` lmao